dropshot/src/server.rs

```diff
@@ -136,67 +170,134 @@ impl<C: ServerContext> HttpServerStarter<C> {
-            }
-            None => {
-                let (starter, app_state, local_addr) =
-                    InnerHttpServerStarter::new(
-                        config,
-                        server_config,
-                        api,
-                        private,
-                        log,
-                        handler_waitgroup.worker(),
-                    )?;
-                HttpServerStarter {
-                    app_state,
-                    local_addr,
-                    wrapped: WrappedHttpServerStarter::Http(starter),
-                    handler_waitgroup,
-                }
-            }
-        };
+        let app_state = Arc::new(DropshotState {
+            private,
+            config: server_config,
+            router: api.into_router(),
+            log: log.clone(),
+            local_addr,
+            tls_acceptor: tls_acceptor.clone(),
+            handler_waitgroup_worker: DebugIgnore(handler_waitgroup.worker()),
+        });
 
-        for (path, method, _) in &starter.app_state.router {
-            debug!(starter.app_state.log, "registered endpoint";
+        for (path, method, _) in &app_state.router {
+            debug!(&log, "registered endpoint";
                 "method" => &method,
                 "path" => &path
             );
         }
 
-        Ok(starter)
+        let http_acceptor = HttpAcceptor { tcp, log: log.clone() };
+
+        Ok(HttpServerStarter {
+            app_state,
+            local_addr,
+            handler_waitgroup,
+            http_acceptor,
+            tls_acceptor,
+        })
     }
 
     pub fn start(self) -> HttpServer<C> {
-        let (tx, rx) = tokio::sync::oneshot::channel::<()>();
-        let log_close = self.app_state.log.new(o!());
-        let join_handle = match self.wrapped {
-            WrappedHttpServerStarter::Http(http) => http.start(rx, log_close),
-            WrappedHttpServerStarter::Https(https) => {
-                https.start(rx, log_close)
-            }
-        };
-        info!(self.app_state.log, "listening");
+        let HttpServerStarter {
+            app_state,
+            local_addr,
+            handler_waitgroup,
+            tls_acceptor,
+            http_acceptor,
+        } = self;
+
+        let (tx, mut rx) = tokio::sync::oneshot::channel::<()>();
+        let make_service = ServerConnectionHandler::new(Arc::clone(&app_state));
+        let log = &app_state.log;
+        let log_close = log.clone();
+        let join_handle = tokio::spawn(async move {
+            use hyper_util::rt::{TokioExecutor, TokioIo, TokioTimer};
+            use hyper_util::server::conn::auto;
+
+            let mut builder = auto::Builder::new(TokioExecutor::new());
+            // http/1 settings
+            builder.http1().timer(TokioTimer::new());
+            // XXX-dap previously, the TLS one did NOT do this http2 step
+            // http/2 settings
+            builder.http2().timer(TokioTimer::new());
```

Comment:
This one I'm not sure about.  Asked in https://github.com/oxidecomputer/dropshot/pull/1028#discussion_r1777995213.